### PR TITLE
feat: add reduceTransparency

### PR DESCRIPTION
**Pull request by rerorero**
See https://github.com/appium/appium/issues/17905

This PR adds appium:reduceTransparency to caps that allows to change `Reduce Transparency` preference.

### CODE REVIEW

File: lib/driver.js [hunk 1]
```diff
@@ -1334,6 +1338,20 @@ class XCUITestDriver extends BaseDriver {
     await this.updateSettings({reduceMotion: isEnabled});
   }
 
+  /**
+   * Set reduceTransparency as 'isEnabled' only when the capabilities has 'reduceTransparency'
+   * The call is ignored for real devices.
+   * @param {?boolean} isEnabled Wether enable reduceTransparency
+   */
+  async setReduceTransparency (isEnabled) {
```
Comment (mykola-mokhnach) [inline lib/driver.js:1346]: I don't think we need a separate driver method for that. Simply call `await this.opts.device.setReduceTransparency(this.opts.reduceTransparency);` in the condition above

File: test/functional/device/accessibility-e2e-specs.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      hasGeneralTab = true;
+    }
+    return hasGeneralTab;
+  }
+
+  describe('ReduceMotion', function() {
+    async function getReduceMotion(driver) {
+      const motionButtonCalledReduceMotion = await showAccessibilityTab(driver);
+      await driver
+        .$(`${PREDICATE_SEARCH}:type == 'XCUIElementTypeCell' AND name == '${motionButtonCalledReduceMotion ? 'Reduce Motion' : 'Motion'}'`)
```
Comment (mykola-mokhnach) [inline test/functional/device/accessibility-e2e-specs.js:50]: this line seems too long to me

Comment (rerorero) [reply test/functional/device/accessibility-e2e-specs.js]: fixed https://github.com/appium/appium-xcuitest-driver/pull/1464/commits/ae128f54446004e7fb5c67e85d1aa613b8128265

File: lib/driver.js (hunk 1 above)
Comment (rerorero) [reply lib/driver.js]: fixed https://github.com/appium/appium-xcuitest-driver/pull/1464/commits/86120049dc69b3d7669c8bdfacf8c55b86464481

File: lib/driver.js [hunk 3]
```diff
@@ -499,6 +499,11 @@ class XCUITestDriver extends BaseDriver {
       await this.setReduceMotion(this.opts.reduceMotion);
     }
 
+    if (_.isBoolean(this.opts.reduceTransparency) && !this.isRealDevice()) {
```
Comment (mykola-mokhnach) [inline lib/driver.js:502]: consider using isSimulator condition instead

Comment (rerorero) [reply lib/driver.js:502]: oops I missed that 👁️

Comment (rerorero) [reply lib/driver.js]: fixed https://github.com/appium/appium-xcuitest-driver/pull/1464/commits/d6b520cef918bc2483ae874422dc598b660337cb

**Comment by rerorero** [comment]
CI started to fail to install WDA 🤔 Will take a look
https://dev.azure.com/AppiumCI/Appium%20CI/_build/results?buildId=24254&view=logs&j=93ef715f-f255-5a46-39ba-d0aa73a4eb56&t=c564e291-e422-500b-9462-249c04740063&l=8815

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**